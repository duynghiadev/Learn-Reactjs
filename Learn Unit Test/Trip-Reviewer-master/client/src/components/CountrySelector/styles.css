.country-selector-container {
  position: relative;
  width: 80%;
  margin-left: 10%;
}

.country-selector-container .c-select {
  width: 100%;
  text-align: left;
  cursor: pointer;
  font-size: 20px;
  border-radius: 3px;
  border: 1px solid #ebebeb;
  box-sizing: border-box;
  padding: 11px 31px;
  background: #f7f7f7;
  font-weight: 300;
}

.country-selector-container .c-dropdown-menu {
  position: absolute;
  z-index: 10;
  top: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
  background-color: #fff;
  box-shadow: 1px 3px 15px black;
}

.country-selector-container .c-dropdown-input {
  width: 100%;
  font-size: 20px;
  border-radius: 3px;
  border: 1px solid #ebebeb;
  box-sizing: border-box;
  padding: 11px 31px;
  background: #f7f7f7;
  font-weight: 300;
}

.country-selector-container .c-dropdown-ul {
  width: 100%;
  padding: 0;
  margin: 0;
}

.country-selector-container .c-dropdown-item {
  padding: 10px 15px;
  cursor: pointer;
  text-align: left;
}

.country-selector-container .c-dropdown-flag {
  margin: 5px 10px 0 0;
}

.country-selector-container .c-dropdown-menu-overflow {
  max-height: 220px;
  overflow: hidden;
  overflow-y: auto;
  border-radius: 3px;
  border: 1px solid #ebebeb;
}

.country-selector-container .dropdown-item:hover {
  background-color: #f7f7f7;
}
