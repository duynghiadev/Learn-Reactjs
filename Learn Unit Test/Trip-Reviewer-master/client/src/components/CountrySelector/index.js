import React from 'react';
import PropTypes from 'prop-types';

import { COUNTRIES } from '../../constants/countries';

import './styles.css';

class CountrySelector extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displayedCountries: COUNTRIES,
      isListVisible: false,
      currentCountry: this.getCountry(props),
    };
  }

  componentWillReceiveProps(nextProps) {
    const newCountry = this.getCountry(nextProps);
    const { currentCountry } = this.state;

    if (newCountry.countryName !== currentCountry.countryName) {
      this.setState({ currentCountry: newCountry });
    }
  }

  componentWillUpdate(nextProps, nextState) {
    const { currentCountry } = this.state;
    const { currentCountry: newCountry } = nextState;

    if (currentCountry.countryName !== newCountry.countryName) {
      this.props.getSelectedCountry(newCountry);
    }
  }

  getCountry = props => {
    let country = {};

    if (props.defaultCountry) {
      country = COUNTRIES.find(item => item.countryName === props.defaultCountry);
    } else if (props.defaultISOALPHA2Code) {
      country = COUNTRIES.find(item => item.ISOALPHA2Code === props.defaultISOALPHA2Code);
    } else if (props.defaultISOALPHA3Code) {
      country = COUNTRIES.find(item => item.ISOALPHA3Code === props.defaultISOALPHA3Code);
    } else if (props.defaultISONumericalCode) {
      country = COUNTRIES.find(item => item.ISONumericalCode === props.defaultISONumericalCode);
    }

    return country && Object.keys(country).length ? country : COUNTRIES[0];
  };

  handleSelectCountry = country => {
    this.setState({ currentCountry: country });
  };

  handleChangeCountry = e => {
    const query = e.target.value.toLowerCase().trim();

    const displayedCountries = COUNTRIES.filter(
      item => item.countryName.toLowerCase().search(query) >= 0
    );

    this.setState({ displayedCountries });
  };

  focusInput = component => {
    if (component) {
      component.focus();
    }
  };

  show = e => {
    this.setState({ isListVisible: true });

    if (typeof window !== 'undefined' && e.target.tagName.toUpperCase() !== 'INPUT') {
      document.addEventListener('click', this.hide);
    }
  };

  hide = e => {
    if (typeof window !== 'undefined' && e.target.tagName.toUpperCase() !== 'INPUT') {
      this.setState({ isListVisible: false });
      document.removeEventListener('click', this.hide);
    }
  };

  render() {
    const { currentCountry, isListVisible, displayedCountries } = this.state;

    return (
      <div className="country-selector-container">
        <div className="dropdown">
          <div className="btn dropdown-toggle c-select" onClick={this.show}>
            <span>
              <div
                alt="flag"
                className={
                  currentCountry.ISOALPHA2Code
                    ? `flag ${currentCountry.ISOALPHA2Code.toLowerCase()} fnone c-dropdown-flag`
                    : 'c-dropdown-flag'
                }
              />
              {currentCountry.countryName}
            </span>
          </div>
          <div
            className="dropdown-menu c-dropdown-menu"
            style={isListVisible ? { display: 'block' } : { display: 'none' }}
          >
            <input
              className="c-dropdown-input"
              type="text"
              ref={component => this.focusInput(component)}
              onInput={this.handleChangeCountry}
            />
            <div className="c-dropdown-menu-overflow">
              <ul className="c-dropdown-ul">
                {displayedCountries.map((item, index) => {
                  const flag = item.ISOALPHA2Code.toLowerCase();
                  return (
                    <li
                      tabIndex={index}
                      className="dropdown-item c-dropdown-item"
                      data-country={item.ISOALPHA2Code}
                      key={item.ISOALPHA2Code}
                      onClick={() => this.handleSelectCountry(item)}
                    >
                      <div data-option={`${item.ISOALPHA2Code}`}>
                        <div className={`flag ${flag} fnone c-dropdown-flag`} /> {item.countryName}
                      </div>
                    </li>
                  );
                })}
              </ul>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CountrySelector.propTypes = {
  defaultCountry: PropTypes.string, // eslint-disable-line react/no-unused-prop-types, max-len, - it is a bug of eslint
  defaultISOALPHA2Code: PropTypes.string, // eslint-disable-line react/no-unused-prop-types, max-len, - it is a bug of eslint
  defaultISOALPHA3Code: PropTypes.string, // eslint-disable-line react/no-unused-prop-types, max-len, - it is a bug of eslint
  defaultISONumericalCode: PropTypes.number, // eslint-disable-line react/no-unused-prop-types, max-len, - it is a bug of eslint
  getSelectedCountry: PropTypes.func.isRequired,
};

CountrySelector.defaultProps = {
  defaultCountry: '',
  defaultISOALPHA2Code: '',
  defaultISOALPHA3Code: '',
  defaultISONumericalCode: null,
};

export default CountrySelector;
