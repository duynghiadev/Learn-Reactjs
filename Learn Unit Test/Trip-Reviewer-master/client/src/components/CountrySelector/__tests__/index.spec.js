import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import CountrySelector from '..';

jest.mock('../../../constants/countries', () => ({
  COUNTRIES: [
    {
      countryName: 'Afghanistan',
      ISOALPHA2Code: 'AF',
      ISOALPHA3Code: 'AFG',
      ISONumericalCode: 4,
    },
    {
      countryName: 'Aland Islands',
      ISOALPHA2Code: 'AX',
      ISOALPHA3Code: 'ALA',
      ISONumericalCode: 248,
    },
    {
      countryName: 'Albania',
      ISOALPHA2Code: 'AL',
      ISOALPHA3Code: 'ALB',
      ISONumericalCode: 8,
    },
  ],
}));

const mockComponent = props => <CountrySelector {...props} />;

describe('<CountrySelector />', () => {
  it('should render component with no props provided', () => {
    const props = {};
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with defaultCountry provided', () => {
    const props = {
      defaultCountry: 'Afghanistan',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with defaultISOALPHA2Code provided', () => {
    const props = {
      defaultISOALPHA2Code: 'AF',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with defaultISOALPHA3Code provided', () => {
    const props = {
      defaultISOALPHA3Code: 'AFG',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with defaultISONumericalCode provided', () => {
    const props = {
      defaultISONumericalCode: 4,
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with a wrong defaultCountry provided', () => {
    const props = {
      defaultCountry: 'This is a test',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('getCountry should return correct value for the wrong input', () => {
    const props = {
      getSelectedCountry: () => {},
    };
    const param = 'THIS IS WRONG';
    const expected = {
      countryName: 'Afghanistan',
      ISOALPHA2Code: 'AF',
      ISOALPHA3Code: 'AFG',
      ISONumericalCode: 4,
    };
    const instance = shallow(mockComponent(props)).instance();

    const result = instance.getCountry(param);

    expect(result).toEqual(expected);
  });

  it('should set state correctly when componentWillReceiveProps is called whith new props', () => {
    const props = {
      defaultCountry: 'Afghanistan',
      getSelectedCountry: () => {},
    };
    const nextProps = {
      defaultCountry: 'Aland Islands',
      getSelectedCountry: () => {},
    };
    const expected = {
      countryName: 'Aland Islands',
      ISOALPHA2Code: 'AX',
      ISOALPHA3Code: 'ALA',
      ISONumericalCode: 248,
    };
    const instance = shallow(mockComponent(props)).instance();

    instance.componentWillReceiveProps(nextProps);

    expect(instance.state.currentCountry).toEqual(expected);
  });

  it('should not set state correctly when componentWillReceiveProps is called whith the same props', () => {
    const setState = jest.fn();
    const props = {
      defaultCountry: 'Afghanistan',
      getSelectedCountry: () => {},
    };
    const nextProps = {
      defaultCountry: 'Afghanistan',
      getSelectedCountry: () => {},
    };
    const instance = shallow(mockComponent(props)).instance();
    instance.setState = setState;

    instance.componentWillReceiveProps(nextProps);

    expect(setState).not.toHaveBeenCalled();
  });

  it('should not call getSelectedCountry when componentWillUpdate is called whith the same country', () => {
    const getSelectedCountry = jest.fn();
    const props = {
      defaultCountry: 'Afghanistan',
      getSelectedCountry,
    };
    const nextState = {
      currentCountry: {
        countryName: 'Afghanistan',
      },
    };
    const instance = shallow(mockComponent(props)).instance();

    instance.componentWillUpdate(props, nextState);

    expect(getSelectedCountry).not.toHaveBeenCalled();
  });

  it('should set state correctly when handleChangeCountry is called', () => {
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        value: '  AfG  ',
      },
    };
    const expected = {
      countryName: 'Afghanistan',
      ISOALPHA2Code: 'AF',
      ISOALPHA3Code: 'AFG',
      ISONumericalCode: 4,
    };
    const instance = shallow(mockComponent(props)).instance();

    instance.handleChangeCountry(event);

    expect(instance.state.displayedCountries).toEqual([expected]);
  });

  it('should set state correctly when handleSelectCountry is called', () => {
    const props = {
      getSelectedCountry: () => {},
    };
    const country = { test: 'test' };
    const instance = shallow(mockComponent(props)).instance();

    instance.handleSelectCountry(country);

    expect(instance.state.currentCountry).toEqual(country);
  });

  it('should call component.focus() when focusInput is called whith valid component', () => {
    const focus = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const component = { focus };
    const instance = shallow(mockComponent(props)).instance();

    instance.focusInput(component);

    expect(focus).toHaveBeenCalled();
  });

  it('should set state and call addEventListener when show is called with correct tagName', () => {
    const addEventListener = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        tagName: 'test',
      },
    };
    const instance = shallow(mockComponent(props)).instance();
    global.document.addEventListener = addEventListener;
    instance.state.isListVisible = false;

    instance.show(event);

    expect(instance.state.isListVisible).toEqual(true);
    expect(addEventListener).toHaveBeenCalledWith('click', instance.hide);
  });

  it('should set state and  not call addEventListener when show is called with wrong tagName', () => {
    const addEventListener = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        tagName: 'input',
      },
    };
    const instance = shallow(mockComponent(props)).instance();
    global.document.addEventListener = addEventListener;
    instance.state.isListVisible = false;

    instance.show(event);

    expect(instance.state.isListVisible).toEqual(true);
    expect(addEventListener).not.toHaveBeenCalledWith('click', instance.hide);
  });

  it('should set state and call removeEventListener when hide is called with correct tagName', () => {
    const removeEventListener = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        tagName: 'test',
      },
    };
    const instance = shallow(mockComponent(props)).instance();
    global.document.removeEventListener = removeEventListener;
    instance.state.isListVisible = true;

    instance.hide(event);

    expect(instance.state.isListVisible).toEqual(false);
    expect(removeEventListener).toHaveBeenCalledWith('click', instance.hide);
  });

  it('should not set state and not call removeEventListener when hide is called with wrong tagName', () => {
    const removeEventListener = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        tagName: 'input',
      },
    };
    const instance = shallow(mockComponent(props)).instance();
    global.document.removeEventListener = removeEventListener;
    instance.state.isListVisible = true;

    instance.hide(event);

    expect(instance.state.isListVisible).toEqual(true);
    expect(removeEventListener).not.toHaveBeenCalledWith('click', instance.hide);
  });

  it('should call handleSelectCountry when country <li> element is clicked', () => {
    const handleSelectCountry = jest.fn();
    const props = {
      getSelectedCountry: () => {},
    };
    const event = {
      target: {
        tagName: 'test',
      },
    };
    const arg = {
      countryName: 'Afghanistan',
      ISOALPHA2Code: 'AF',
      ISOALPHA3Code: 'AFG',
      ISONumericalCode: 4,
    };
    const node = shallow(mockComponent(props));
    const instance = node.instance();

    instance.handleSelectCountry = handleSelectCountry;
    instance.show(event);

    node.find('[data-country="AF"]').simulate('click');

    expect(handleSelectCountry).toHaveBeenCalledWith(arg);
  });
});
