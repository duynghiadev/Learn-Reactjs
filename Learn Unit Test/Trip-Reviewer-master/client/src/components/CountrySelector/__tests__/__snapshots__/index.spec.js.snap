// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountrySelector /> should render component with a wrong defaultCountry provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CountrySelector /> should render component with defaultCountry provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CountrySelector /> should render component with defaultISOALPHA2Code provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CountrySelector /> should render component with defaultISOALPHA3Code provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CountrySelector /> should render component with defaultISONumericalCode provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CountrySelector /> should render component with no props provided 1`] = `
<div
  className="country-selector-container"
>
  <div
    className="dropdown"
  >
    <div
      className="btn dropdown-toggle c-select"
      onClick={[Function]}
    >
      <span>
        <div
          alt="flag"
          className="flag af fnone c-dropdown-flag"
        />
        Afghanistan
      </span>
    </div>
    <div
      className="dropdown-menu c-dropdown-menu"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <input
        className="c-dropdown-input"
        onInput={[Function]}
        type="text"
      />
      <div
        className="c-dropdown-menu-overflow"
      >
        <ul
          className="c-dropdown-ul"
        >
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AF"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              data-option="AF"
            >
              <div
                className="flag af fnone c-dropdown-flag"
              />
               
              Afghanistan
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AX"
            onClick={[Function]}
            tabIndex={1}
          >
            <div
              data-option="AX"
            >
              <div
                className="flag ax fnone c-dropdown-flag"
              />
               
              Aland Islands
            </div>
          </li>
          <li
            className="dropdown-item c-dropdown-item"
            data-country="AL"
            onClick={[Function]}
            tabIndex={2}
          >
            <div
              data-option="AL"
            >
              <div
                className="flag al fnone c-dropdown-flag"
              />
               
              Albania
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
